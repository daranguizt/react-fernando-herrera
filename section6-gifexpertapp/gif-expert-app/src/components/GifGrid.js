import React, { useState, useEffect, Fragment } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { getGifs } from "../helpers/getGifs";
import GifGridItem from "./GifGridItem";

const GifGrid = ({ category }) => {
  //   const [images, setImages] = useState([]);

  //   useEffect(() => {
  //     getGifs(category).then(setImages);
  //   }, [category]);

  const { data, loading } = useFetchGifs(category);

  return (
    <Fragment>
      <h3 className="title">{category}</h3>
      {loading && <p >Loading...</p>}
      <div className="results">
        {
          data.map((item) => (
            <GifGridItem key={item.id} title={item.title} url={item.url} />
          ))}
      </div>
    </Fragment>
  );
};

export default GifGrid;
