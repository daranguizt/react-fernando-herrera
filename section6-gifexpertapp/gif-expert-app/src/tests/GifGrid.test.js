import { shallow } from "enzyme";
import GifGrid from "../components/GifGrid";
import { useFetchGifs } from "../hooks/useFetchGifs";
jest.mock("../hooks/useFetchGifs");

describe("tests on gifgrid", () => {
  const category = "test";

  test("should show and match snapshot", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show items when images are loaded via useFetchGifs", () => {
    const gifs = [
      {
        id: 123,
        url: "https://asd.net",
        title: "test",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
