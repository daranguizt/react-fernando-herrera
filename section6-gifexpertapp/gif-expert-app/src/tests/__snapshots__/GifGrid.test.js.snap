// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on gifgrid should show and match snapshot 1`] = `
<Fragment>
  <h3
    className="title"
  >
    test
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="results"
  />
</Fragment>
`;

exports[`tests on gifgrid should show items when images are loaded via useFetchGifs 1`] = `
<Fragment>
  <h3
    className="title"
  >
    test
  </h3>
  <div
    className="results"
  >
    <GifGridItem
      key="123"
      title="test"
      url="https://asd.net"
    />
  </div>
</Fragment>
`;
